### ISSUE TITLE
[Bug]  Throw "Promise Undefine" Error when create a new account 

### ISSUE THREAD

**Issue by LKMMKL** [AzureAD/microsoft-authentication-library-for-dotnet#4825]
### Library version used

4.61.3

### .NET version

Enviroment:
1.  WPF
 2. .NET Framwork 4.7.2

![QDYHX5{L5O2VQR2Z98)W~4J](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/assets/31981596/381bb0cf-5e67-4f18-a935-572683783708)

### Scenario

PublicClient - desktop app

### Is this a new or an existing app?

The app is in production, and I have upgraded to a new version of MSAL

### Issue description and reproduction steps

1. Create new account on the login page.
2. After filling in the region and date of birth，it throw the error "Promise Undefined"

### Relevant code snippets

```csharp
var authResult = await otApp.AcquireTokenInteractive(scopes)
                                    .WithPrompt(Microsoft.Identity.Client.Prompt.SelectAccount)
                                    .ExecuteAsync();
```

### Expected behavior

I hope it create new account success.

### Identity provider

Microsoft Entra ID (Work and School accounts and Personal Microsoft accounts)

**Comment by localden** [comment]
@LKMMKL - the error is odd, and doesn't seem like it's caused by MSAL directly. Arkose Labs _seems_ to be the vendor that provides the "Verify you're not a bot" experiences that get embedded in various flows. The error message obstrcuts the underlying view, but was that by chance what you were getting? We might need to raise this with the service team.

**Comment by bgavrilMS** [comment]
This is probably due to the use of WebView1 (embedded browser). It has a very old engine (Internet Explorer!) and modern javascript will error out. Most of the auth screens are ok, but not all of them. If you are using 3rd party MFA provider, they probably don't test with WebView1. 

Solution is to move to WAM: https://aka.ms/msal-net-wam

**Comment by localden** [comment]
I don't even think it's a third-party MFA, this is likely the embedded puzzle (screenshot from BleepingComputer): 

![Puzzle rendered during MSA authentication](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/assets/53200638/8421784c-85af-4701-9aee-34feb8f583df)

**Comment by LKMMKL** [comment]
@localden  It show “Help us defeat robots”
![V X`8AKVZED`MV(_RLAL}CI](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/assets/31981596/70cbe69c-99bb-4974-a772-5f1a07c6f174)

**Comment by LKMMKL** [comment]
> This is probably due to the use of WebView1 (embedded browser). It has a very old engine (Internet Explorer!) and modern javascript will error out. Most of the auth screens are ok, but not all of them. If you are using 3rd party MFA provider, they probably don't test with WebView1.
> 
> Solution is to move to WAM: https://aka.ms/msal-net-wam

Thanks! WAM has solved the problem to some extent, but it cannot operate in the administrator process. Do you have any suggestions.

**Comment by bgavrilMS** [comment]
This is strange. Are you using the latest version of MSAL ? Admin based scenarios are supported.

**Comment by localden** [comment]
I think we need to raise the issue with the script with the service team, because the current flow seems to be breaking _because_ of custom anti-fraud verification.

That being said, when I try to use WAM in an application with admin permissions, I am getting an error:

![Error shown when WAM is ran with admin permissions](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/assets/53200638/8459dbc4-b580-4261-9f81-de0fa702e67c)

Will be checking with the team if this is a known issue.

**Comment by LKMMKL** [comment]
> 这很奇怪。您使用的是最新版本的 MSAL 吗？支持基于管理员的方案。

Yeah , I use the version "4.61.3", maybe need some special configuration? But I couldn't find something related in the document.

**Comment by LKMMKL** [comment]
@localden 
Thanks, , looking forward to your confirmation. I have received the same error at "4.61.3" in admin  application .

**Comment by localden** [comment]
@LKMMKL can you please try and update the [`Microsoft.Identity.Client.NativeInterop`](https://www.nuget.org/packages/Microsoft.Identity.Client.NativeInterop/) to the latest version (0.16.2) and see if the issue still reproduces?

**Comment by LKMMKL** [comment]
@localden 
Are you referring to “ admin application” problem？
I update this library, and I found it work if the window is cmd window,.
For uwp window, the problem still exists.
`   
IntPtr handle = FindWindow(null, "UWPWIndow");
                var authResult = await PublicClientApp.AcquireTokenInteractive(scopes)
                     .WithParentActivityOrWindow(handle) // optional, used to center the browser on the window
                    .WithPrompt(Prompt.Consent)
                    .WithPrompt(Prompt.ForceLogin)
                    .ExecuteAsync();
`

**Comment by LKMMKL** [comment]
May be you nedd a small demo?
In the demo, I try to show the page on uwp window with **WithParentActivityOrWindow** in wpf application.
Run Wpf with Administrator privileges，It will throw "User canceled authentication".
[Uploading SmallDemo.zip…]()

**Comment by LKMMKL** [comment]
Sorry, the file was not successfully uploaded yesterday。
My tech stack is Wpf and Uwp, Wpf  as a backend service, the code run in wpf, and  I want the login page to be displayed on UWP.
When WPF is started with normal user privileges,  it works.
When WPF is started with administrator privileges, it doesn't work.
Does [deprecated support for UWP](https://devblogs.microsoft.com/identity/uwp-xamarin-msal-net-deprecation/) have an impact on this situation, as the code is not running on UWP？
[AppUwp.zip](https://github.com/user-attachments/files/16276459/AppUwp.zip)
[WpfClient.zip](https://github.com/user-attachments/files/16276464/WpfClient.zip)

**Comment by localden** [comment]
This seems like a special case - you're crossing the WPF and UWP boundaries. Can you tell us more about the scenario you're trying to implement here? Something could be funky with permission boundaries between WPF and UWP when running with admin permissions.

@ashok672 - did you see a scenario like this before?

**Comment by LKMMKL** [comment]
My pleasure to explain my current usage scenario to you.
UWP is our main program, and WPF is a windowless backend service that needs to be started with administrator privileges because it requires operating some system resources.
Since there are no other windows, we can only choose to display the login page on the UWP window.
Of course, if this problem cannot be solved, I may choose to launch it with normal privileges and use other programs through communication to solve the issue of operating system resources, but this may require significant changes for us.

**Comment by localden** [comment]
I guess the question is - why UWP window and not a WPF window? We're still investigating the scenario, but I am curious as to why not keep everything on one stack.

**Comment by LKMMKL** [comment]
Due to project reasons, it may be difficult to make some changes.
